File: protocol/bin/datahandlers/ExampleHandler.java
import org.apache.log4j.Logger;
import org.vertx.java.core.Handler;
import org.vertx.java.core.eventbus.EventBus;
import org.vertx.java.core.eventbus.Message;
import org.vertx.java.core.json.JsonObject;

import org.vertx.java.platform.Verticle;

import com.splunk.modinput.ModularInput;
import com.splunk.modinput.Stream;

public class ExampleHandler extends Verticle {

	private static Logger logger = Logger.getLogger(ExampleHandler.class);

	public void start() {

		JsonObject config = container.config();
		final String stanza = config.getString("stanza");
		String eventBusAddress = config.getString("address");
		EventBus eb = vertx.eventBus();

		Handler<Message<byte[]>> myHandler = new Handler<Message<byte[]>>() {
			public void handle(Message<byte[]> message) {

				try {
				    
				    //actual received data
					byte[] data = message.body();

                    //pre-processing / transforming the received data
					String output = "Ignoring received data and replacing with JAVA GOATS!!!!!!";

					//bundle up String output and send to Splunk
					Stream stream = HandlerUtil.getStream(output, stanza);
					ModularInput.marshallObjectToXML(stream);

				} catch (Exception e) {
					logger.error("Error writing received data: "
							+ ModularInput.getStackTrace(e));
				}

			}
		};

		eb.registerHandler(eventBusAddress, myHandler);

	}

}

File: protocol/bin/datahandlers/ExampleHandler.js
var eb = require('vertx/event_bus');
var container = require('vertx/container');
var console = require('vertx/console');

var config = container.config;


var eventBusAddress = config.address;
var outputAddress = config.output_address;

var myHandler = function(message) {
	//actual received data
	var data = message;
	
	//pre-processing / transforming the received data
	var output = "Ignoring received data and replacing with Javascript GOATS!!!!!!";

	//pass along to output handler
	eb.send(outputAddress,output);
	
}

eb.registerHandler(eventBusAddress, myHandler);